src/macros.rs

```diff
@@ -446,6 +446,43 @@ macro_rules! service {
             }
         }
 
+        #[allow(non_camel_case_types)]
+        impl<tarpc_service_S__> $crate::tokio_service::NewService
+            for tarpc_service_AsyncServer__<tarpc_service_S__>
+            where tarpc_service_S__: FutureService
+        {
+            type Request = <Self as $crate::tokio_service::Service>::Request;
+            type Response = <Self as $crate::tokio_service::Service>::Response;
+            type Error = <Self as $crate::tokio_service::Service>::Error;
+            type Instance = Self;
+
+            fn new_service(&self) -> ::std::io::Result<Self> {
+                Ok(self.clone())
+            }
+        }
+
+        /// The future returned by `FutureServiceExt::listen`.
+        #[allow(unused)]
+        pub struct Listen<S>
+            where S: FutureService,
+        {
+            inner: $crate::server::Listen<tarpc_service_AsyncServer__<S>,
+                                          tarpc_service_Request__,
+                                          tarpc_service_Response__,
+                                          tarpc_service_Error__>,
+        }
+
+        impl<S> $crate::futures::Future for Listen<S>
+            where S: FutureService
+        {
+            type Item = ();
+            type Error = ();
+
+            fn poll(&mut self) -> $crate::futures::Poll<(), ()> {
+                self.inner.poll()
+            }
+        }
+
         /// Provides a function for starting the service. This is a separate trait from
         /// `FutureService` to prevent collisions with the names of RPCs.
         pub trait FutureServiceExt: FutureService {
```

Comment:
Can you update the comments for this? It says stuff about running on the default tokio loop, which I believe is incorrect now, since the caller has to do the running.